test-site/src/components/MapPin.tsx

```diff
@@ -1,9 +1,48 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
+import { MapPopup } from './MapPopup';
+
+interface MapPinProps {
+  result: Result<Location>
+  index: number
+  mapbox: Map
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps) => {
+  const popupRef = useRef(new Popup({ offset: 15 }));
+  const initialRender = useRef(true);
+  const [active, setActive] = useState(false);
+
+  useEffect(() => {
+    if (!initialRender.current && active) {
+      const popup = document.createElement('div');
+      ReactDOM.render(<MapPopup entity={props.result} />, popup);
```

Comment:
I feel like we might not need this MapPopup component if it's literally just a div? Maybe we could just call setText instead and then skip out on the ReactDOM.render?